rpcserver.go

```diff
@@ -5067,3 +5075,108 @@ func (r *rpcServer) SubscribeChannelBackups(req *lnrpc.ChannelBackupSubscription
 		}
 	}
 }
+
+// ChannelAcceptor dispatches a bi-directional streaming RPC in which
+// OpenChannel requests are sent to the client and the client responds with
+// a boolean that tells LND whether or not to accept the channel. This allows
+// node operators to specify their own criteria for accepting inbound channels
+// through a single persistent connection. There may only be one client at a
+// time for this RPC.
+func (r *rpcServer) ChannelAcceptor(stream lnrpc.Lightning_ChannelAcceptorServer) error {
+
+	var (
+		acceptorClient *chanacceptor.Client
+		rpcAcceptor    *chanacceptor.RPCAcceptor
+		ok             bool
+	)
+
+	// Locate the RPCAcceptor within the ChainedAcceptor and register this RPC client.
+	openChanPredicate := r.server.fundingMgr.cfg.OpenChannelPredicate
+	chainedAcceptor, _ := openChanPredicate.(*chanacceptor.ChainedAcceptor)
```

Comment:
Whenever a type assertion is done, the second variable should _always_ be checked. 

It's also a code smell that we're accessing the _config_ of the funding manager here in order to access the RPC acceptor. Atm, the rpc acceptor is a singleton, so we could add a package level method to obtain the latest version of the acceptor.